nav2_navfn_planner/src/main.cpp

```diff
@@ -10,17 +10,20 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License. Reserved.
+// limitations under the License.
 
 #include <memory>
-#include "rclcpp/rclcpp.hpp"
+
 #include "nav2_navfn_planner/navfn_planner.hpp"
+#include "rclcpp/rclcpp.hpp"
 
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  rclcpp::spin(std::make_shared<nav2_navfn_planner::NavfnPlanner>());
+  auto node = std::make_shared<nav2_navfn_planner::NavfnPlanner>();
+  rclcpp::spin(node->get_node_base_interface());
```

Comment:
@bpwilcox Sure. Because the rclcpp::spin has two variants:

```C++
void spin(rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_ptr);
void spin(rclcpp::Node::SharedPtr node_ptr);

```
You can either pass it a Node::SharedPtr or a NodeBaseInterface::SharedPtr. In our case, we have a lifecycle node, so we must pass a NodeBaseInterface::SharedPtr, which we get from the lifecycle node via node->get_node_base_interface().

This tends to be a common theme across many ROS2 classes: they take a Node directly, but if you use a LifecycleNode, you have to provide a sub-interface and not the LifeCycleNode::SharedPtr as you might expect; there isn't support in the various classes/methods for simply passing a LifecycleNode::SharedPtr.